integration-cli/docker_cli_cp_test.go

```diff
@@ -136,200 +107,151 @@ func (s *DockerSuite) TestCpRelativePath(c *check.C) {
 
-	dockerCmd(c, "cp", cleanedContainerID+":"+path, tmpdir)
+	dockerCmd(c, "cp", cID+":"+path, tmpdir)
 
 	// We should have copied a symlink *NOT* the file itself!
 	linkTarget, err := os.Readlink(tmpname)
-	if err != nil {
-		c.Fatal(err)
-	}
+	c.Assert(err, checker.IsNil)
 
-	if linkTarget != filepath.FromSlash(cpFullPath) {
-		c.Errorf("symlink target was %q, but expected: %q", linkTarget, cpFullPath)
-	}
+	c.Assert(linkTarget, checker.Equals, filepath.FromSlash(cpFullPath))
 }
 
 // Check that symlinks to a directory behave as expected when copying one from
 // a container.
 func (s *DockerSuite) TestCpFromSymlinkToDirectory(c *check.C) {
 	testRequires(c, DaemonIsLinux)
-	out, exitCode := dockerCmd(c, "run", "-d", "busybox", "/bin/sh", "-c", "mkdir -p '"+cpTestPath+"' && echo -n '"+cpContainerContents+"' > "+cpFullPath+" && ln -s "+cpTestPathParent+" /dir_link")
-	if exitCode != 0 {
-		c.Fatal("failed to create a container", out)
-	}
+	out, _ := dockerCmd(c, "run", "-d", "busybox", "/bin/sh", "-c", "mkdir -p '"+cpTestPath+"' && echo -n '"+cpContainerContents+"' > "+cpFullPath+" && ln -s "+cpTestPathParent+" /dir_link")
 
-	cleanedContainerID := strings.TrimSpace(out)
+	cID := strings.TrimSpace(out)
 
-	out, _ = dockerCmd(c, "wait", cleanedContainerID)
-	if strings.TrimSpace(out) != "0" {
-		c.Fatal("failed to set up container", out)
-	}
+	out, _ = dockerCmd(c, "wait", cID)
+	// failed to set up container
+	c.Assert(strings.TrimSpace(out), checker.Equals, "0")
 
 	testDir, err := ioutil.TempDir("", "test-cp-from-symlink-to-dir-")
-	if err != nil {
-		c.Fatal(err)
-	}
+	c.Assert(err, checker.IsNil)
 	defer os.RemoveAll(testDir)
 
 	// This copy command should copy the symlink, not the target, into the
 	// temporary directory.
-	dockerCmd(c, "cp", cleanedContainerID+":"+"/dir_link", testDir)
+	dockerCmd(c, "cp", cID+":"+"/dir_link", testDir)
 
 	expectedPath := filepath.Join(testDir, "dir_link")
 	linkTarget, err := os.Readlink(expectedPath)
-	if err != nil {
-		c.Fatalf("unable to read symlink at %q: %v", expectedPath, err)
-	}
+	c.Assert(err, checker.IsNil)
 
-	if linkTarget != filepath.FromSlash(cpTestPathParent) {
-		c.Errorf("symlink target was %q, but expected: %q", linkTarget, cpTestPathParent)
-	}
+	c.Assert(linkTarget, checker.Equals, filepath.FromSlash(cpTestPathParent))
 
 	os.Remove(expectedPath)
 
 	// This copy command should resolve the symlink (note the trailing
 	// separator), copying the target into the temporary directory.
-	dockerCmd(c, "cp", cleanedContainerID+":"+"/dir_link/", testDir)
+	dockerCmd(c, "cp", cID+":"+"/dir_link/", testDir)
 
 	// It *should not* have copied the directory using the target's name, but
 	// used the given name instead.
 	unexpectedPath := filepath.Join(testDir, cpTestPathParent)
-	if stat, err := os.Lstat(unexpectedPath); err == nil {
-		c.Fatalf("target name was copied: %q - %q", stat.Mode(), stat.Name())
+	stat, err := os.Lstat(unexpectedPath)
+	if err == nil {
```

Comment:
I tried that at first, but it has panic when running the test.
Because `stat` is unavailable when err != nil.